### ISSUE TITLE
Question about train_ClsConfSet.lst

### ISSUE THREAD

**Issue by yufei1900** [yzou2/CRST#11]
Hello, very impressive work!
I have a question about the train_ClsConfSet.lst in CRST/dataset/list/cityscapes/train_ClsConfSet.lst. 
In this training list, there are only 505 lines while the whole cityscapes training list should be 2975 lines. Would you mind explaining how do you get this list? Were the results in your paper produced by this subset? Have you ever try what would happen if you use the whole set? 
Thank you in advance~

**Comment by yzou2** [comment]
Thanks a lot for your interest in our work. 

The train_ClsConfSet.lst is a subset of the whole cityscapes training list. It is first used in CBST (https://github.com/yzou2/CBST). The results on CRST is produced by this subset. In our ECCV paper, I tried UDA with the full train-set, but the performance is slightly worse than using the ClsConfSet.

Intuitively, we assume that not all images in Cityscapes are overall good enough for generating pseudo-labels since some images have overall bad predictions/pseudo-labels and lots of false positives. If we can first select these good images, we may have better adaptation performance. Thus we select the subset according to a score called class balanced score (CBS). CBS is defined for each image. The score captures the global prediction confidence for an image, considering the class-balance. For computing CBS, first get the average of pixel-level prediction confidence for each class in one image, then get the mean of the class-wise average confidence. After getting all CBS of cityscapes train-set, we set a threshold to select the ClsConfSet.
